### PR TITLE
New method to ingest subject, session, and scan data. Create jupyter notebooks.  Update readme.

### PR DESCRIPTION
**Pull request by kabilar**

### CODE REVIEW

File: workflow_imaging/pipeline.py [hunk 1]
```diff
@@ -34,6 +34,12 @@ class Session(dj.Manual):
     session_datetime: datetime(3)
     """
 
+    class Directory(dj.Part):
+        definition = """
+        -> master
+        session_dir: varchar(256)       # Absolute path
```
Comment (ttngu207) [inline workflow_imaging/pipeline.py:40]: this should be relative path, relative to the `imaging_root_data_dir`

File: workflow_imaging/paths.py [hunk 2]
```diff
@@ -9,10 +9,8 @@ def get_imaging_root_data_dir():
 
 def get_scan_image_files(scan_key):
     # Folder structure: root / subject / session / .tif (raw)
-    data_dir = get_imaging_root_data_dir()
-    subj_dir = data_dir / scan_key['subject']
-    sess_datetime_string = scan_key['session_datetime'].strftime('%Y%m%d_%H%M%S')
-    sess_dir = subj_dir / sess_datetime_string
+    from workflow_imaging.pipeline import Session
+    sess_dir = pathlib.Path((Session.Directory & scan_key).fetch('session_dir')[0])
```
Comment (ttngu207) [inline workflow_imaging/paths.py:13]: you can do:
```
sess_dir = pathlib.Path((Session.Directory & scan_key).fetch1('session_dir'))
```

Comment (ttngu207) [reply workflow_imaging/paths.py:13]: also, let's do a check if directory path exist and raise error appropriately

File: workflow_imaging/ingest.py [hunk 3]
```diff
@@ -1,68 +1,70 @@
...
+        if loaded_scan is not None:
+            recording_time = get_scanimage_acq_time(loaded_scan)
+            header = parse_scanimage_header(loaded_scan)
+            session_key = {'subject': session['subject'], 'session_datetime': recording_time}
+            scanner = header['SI_imagingSystem'].strip('\'')
+            if session_key not in Session.proj():
+                scanners.append({'scanner': scanner})
+                sessions.append(session_key)
+                scans.append({**session_key, 'scan_id': 0, 'scanner': scanner})
+                session_directories.append({**session_key, 'session_dir': session['session_dir']})
```
Comment (ttngu207) [inline workflow_imaging/ingest.py:46]: this is where we need to extract the relative path of the `session_dir` (relative to the `imaging_root_data_dir`) and only insert the relative path, you can use `.relative_to()` (from `pathlib` library)

File: workflow_imaging/ingest.py [hunk 4]
```diff
@@ -1,68 +1,70 @@
...
-                                   for sc in scan.Scan.fetch('KEY')], skip_duplicates=True)
+    print(f'\n---- Insert {len(scans)} entry(s) into scan.Scan ----')
+    scan.Scan.insert(scans, skip_duplicates=True)
 
-    # caiman - 2D
-    imaging.ProcessingTask.insert([{**sc, 'paramset_idx': 1, 'task_mode': 'load'}
-                                   for sc in (scan.Scan & (scan.ScanInfo & 'ndepths = 1')).fetch('KEY')],
-                                  skip_duplicates=True)
+    print(f'\n---- Insert {len(processing_tasks)} entry(s) into imaging.ProcessingTask ----')
+    imaging.ProcessingTask.insert(processing_tasks, skip_duplicates=True)
```
Comment (ttngu207) [inline workflow_imaging/ingest.py:65]: we really should not need `skip_duplicates=True` here, if there's a duplication then that indicates bug in the code somewhere

**Review by ttngu207 — CHANGES_REQUESTED** [review]
only a few minor suggestions, looks good overall

File: workflow_imaging/ingest.py (hunk 4 above)
Comment (kabilar) [reply workflow_imaging/ingest.py:65]: what about the case where the user appends to `subjects.csv` or `sessions.csv`, and reruns this script

File: workflow_imaging/paths.py (hunk 2 above)
Comment (kabilar) [reply workflow_imaging/paths.py]: thanks, updated

File: workflow_imaging/pipeline.py (hunk 1 above)
Comment (kabilar) [reply workflow_imaging/pipeline.py:40]: i agree that we should store the relative path in the table, but am a bit confused as to why we should store the absolute path in the csv file.  thanks

Comment (ttngu207) [reply workflow_imaging/pipeline.py:40]: I think that offers better usability, usually users would just copy and paste the whole path there, it's a bit much to have to expect from them to relativize the path first before placing into the csv file

File: workflow_imaging/ingest.py (hunk 4 above)
Comment (ttngu207) [reply workflow_imaging/ingest.py:65]: we should already have a safeguard against that, with the `if session_key not in Session.proj()`

Comment (kabilar) [reply workflow_imaging/ingest.py:65]: makes sense, thanks

File: workflow_imaging/pipeline.py (hunk 1 above)
Comment (kabilar) [reply workflow_imaging/pipeline.py:40]: makes sense, thanks

File: workflow_imaging/ingest.py (hunk 3 above)
Comment (kabilar) [reply workflow_imaging/ingest.py:46]: thanks

**Review by ttngu207 — APPROVED** [review]

**Pull request merged**